pkg/relayer/proxy/server_builder.go

```diff
@@ -107,3 +111,36 @@ func (rp *relayerProxy) initializeProxyServers(
 
 	return proxyServers, nil
 }
+
+// waitForSupplierToStake waits in a loop until it gets the on-chain supplier's
```

Comment:
I think we really need to qualify that this is for testing or debugging purposes.

The analogy I think of is a bank account. If we were building a web2 banking sass business, this function would be "waitForUserToCreateBankAccount" before opening up their app. It makes sense, but would almost never be necessary since it's a one-time thing.